### ISSUE TITLE
[Request] More Robust AP Refilling

### ISSUE THREAD

**Issue by JeremyMerkur** [Masuzu/ZooeyBot#994]
Sometimes Zooeybot fails to refill AP and gets stuck. According to #102 this is a deliberate design decision. It would be really nice if there were an optional setting (e.g. ConfirmAPRefill = true) that would check if the AP has been successfully refilled (refresh summon page?), and try again if not. Perhaps with some sort of limit (e.g. MaxAPRefillAttempts = 3). And maybe have it play an alert if it hits the max attempts and still fails?

**Comment by Masuzu** [comment]
Hello,
I will add retries for AP refills, but can't do so before January 5, 2019.

**Comment by JeremyMerkur** [comment]
I noticed that the January 17 update added potion retrying. However, it has not solved my issue. To elaborate, sometimes Zooeybot will try to refill AP, fail, but think it has succeeded (or doesn't think it needs to refill AP), and then get stuck at the following screen:

![failed to use potion](https://user-images.githubusercontent.com/1916550/51504973-05311e00-1db2-11e9-833e-a0a579b5d938.png)

Whatever Zooey does to refresh the summon list doesn't work, as the page does not refresh.

What I think is a related, but opposite situation, is where (both before and after the January 17 update) Zooeybot would use multiple potions, thinking that the earlier requests had failed. I have had, a couple times, my AP completely refilled to 999, and then get stuck there (and other times only use a few potions before continuing).

The common thread, as I see it, is that Zooeybot is not confirming what your AP total is after trying to use a potion. Instead, it seems to guess whether or not the request was successful (presumably from the server response), and take appropriate action from there. But that does not seem to be a reliable method. What I would like us for Zooeybot to double-check my AP total after trying to use  a potion. This would cause a delay (since you'd have to query the server), which is why I suggested making it toggleable (so people who don't have as much of an issue as I do can skip it for better speed).

I hope this assists you, and thank you for your work.
